src/components/settings/ImportAllDialog/index.tsx

```diff
@@ -0,0 +1,250 @@
+  const [error, setError] = useState<string>()
+
+  // Parse the jsonData whenever it changes
+  const { addedSafes, addedSafesCount, addressBook, addressBookEntriesCount } = useGlobalImportJsonParser(jsonData)
+
+  const dispatch = useAppDispatch()
+
+  const onDrop = (acceptedFiles: File[]) => {
+    if (acceptedFiles.length === 0) {
+      return
+    }
+    const file = acceptedFiles[0]
+    const reader = new FileReader()
+    reader.onload = function (evt) {
+      if (!evt.target) {
+        return
+      }
+      if (typeof evt.target.result !== 'string') {
+        return
+      }
+      setJsonData(evt.target.result)
+      setFileName(file.name)
+    }
+    reader.readAsText(file)
+  }
+
+  const onRemove: MouseEventHandler = (event) => {
+    setJsonData(undefined)
+    setFileName(undefined)
+    event.preventDefault()
+    event.stopPropagation()
+  }
+
+  const { getRootProps, getInputProps, isDragActive, isDragAccept, isDragReject } = useDropzone({
+    maxFiles: 1,
+    onDrop,
+    accept: {
+      text: ['.json'],
+    },
+  })
+
+  const handleImport = () => {
+    if (!addressBook && !addedSafes) {
+      return
+    }
+
+    if (addressBook) {
+      dispatch(addressBookSlice.actions.setAddressBook(addressBook))
+    }
+
+    if (addedSafes) {
+      dispatch(addedSafesSlice.actions.setAddedSafes(addedSafes))
+    }
+
+    dispatch(
+      showNotification({
+        variant: 'success',
+        groupKey: 'global-import-success',
+        message: 'Successfully imported data',
+        detailedMessage: `${addedSafesCount > 0 ? `${addedSafesCount} Safes were added. \n` : ''}${
+          addressBookEntriesCount > 0 ? `${addressBookEntriesCount} addresses were added to your address book.` : ''
+        }`,
+      }),
+    )
+
+    trackEvent({
+      ...SETTINGS_EVENTS.DATA.IMPORT_ALL,
+      label: `Added Safes: ${addedSafesCount}, Addressbook entries: ${addressBookEntriesCount}`,
+    })
+
+    handleClose()
+  }
+
+  return (
+    <ModalDialog open onClose={handleClose} dialogTitle="Data Import" hideChainIndicator>
+      <DialogContent>
+        <div>
+          {fileName ? (
+            <Box>
+              <>
+                <Grid container direction="column" gap={1}>
+                  <Grid container gap={1} display="flex" alignItems="center">
+                    <Grid item xs={1}>
+                      <SvgIcon
+                        component={FileIcon}
+                        inheritViewBox
+                        fontSize="small"
+                        color="primary"
+                        sx={{ fill: 'none' }}
+                      />
+                    </Grid>
+                    <Grid item xs={7}>
+                      {fileName}
+                    </Grid>
+
+                    <Grid item xs display="flex" justifyContent="flex-end">
+                      <IconButton onClick={onRemove}>
+                        <HighlightOffIcon color="primary" width={16} height={16} />
```

Comment:
`fontSize="small"` should achieve the same as setting the `width`/`height` manually.